autonomi/src/client/data_types/graph.rs

```diff
@@ -54,7 +54,38 @@ impl Client {
         &self,
         address: GraphEntryAddress,
     ) -> Result<GraphEntry, GraphError> {
-        let graph_entries = self.network.get_graph_entry(address).await?;
+        let result = self.graph_entry_get_inner(address).await;
+
+        // Reporting
+        if let Some(channel) = self.client_event_sender.as_ref() {
+            let xor_name = address.xorname();
+            let event = if result.is_ok() {
+                ClientEvent::DownloadSucceeded(*xor_name)
+            } else {
+                ClientEvent::DownloadFailed(*xor_name)
+            };
+
+            if let Err(err) = channel.send(event).await {
+                error!("Failed to send client event: {err:?}");
+            }
+        }
+        result
```

Comment:
This pattern is for every type. Can't we make this a function so we reduce this to one line?

Reply:
Sounds good! Will try it out!